server/fleet/apple_mdm.go

```diff
@@ -251,6 +252,57 @@ func (mc Mobileconfig) ParseConfigProfile() (*MDMAppleConfigProfile, error) {
 	}, nil
 }
 
+// GetPayloadTypes attempts to parse the PayloadContent list of the Mobileconfig's TopLevel object.
+// It returns the PayloadType for each element of the PayloadContent list.
+//
+// See also https://developer.apple.com/documentation/devicemanagement/toplevel
+func (mc Mobileconfig) GetPayloadTypes() ([]string, error) {
```

Comment:
In regards to this function, the `IsEncrypted` check is really a good idea, why aren't we doing it in `MobileConfig.ParseConfigProfile`?

Which makes me think, can we call `ParseConfigProfile` or maybe receive a parsed config as a parameter and here just loop to get all the `PayloadType`s?

Reply:
After chatting with Sarah: I was confused between the top-level object and the array of payloads that we receive.

Thinking a bit more about this, I think at some point we'll want to store `PayloadContent` and `IsEncrypted` in `MDMAppleConfigProfile` (not necessarily in the db!) so this is more clearly something like:

```go
parsed, err := mc.ParseConfigProfile()
if parsed.PayloadType != "Configuration" {
	return nil, fmt.Errorf("invalid PayloadType: %s", parsed.PayloadType)
}

if len(parsed.PayloadContent) < 1 {
	if parsed.IsEncrypted {
		return nil, errors.New("encrypted PayloadContent")
	}
	return nil, errors.New("empty PayloadContent")
}
```

Reply:
note that I'm not asking to do this ☝️ in this PR